### PR TITLE
Pin requests < 2.32.0

### PR DESCRIPTION
**Pull request by jmsanders**
While we investigate breaking docker tests:

https://buildkite.com/dagster/dagster-dagster/builds/83562

### CODE REVIEW

File: python_modules/dagster/setup.py [hunk 1]
```diff
@@ -96,7 +96,7 @@ def get_version() -> str:
         "python-dateutil",
         "python-dotenv",
         "pytz",
-        "requests",
+        "requests<2.32.0",  # 2.32.0 breaks our docker tests https://buildkite.com/dagster/dagster-dagster/builds/83562
```
Comment (alangenfeld) [inline python_modules/dagster/setup.py:99]: would it be better to put this pin this in dagster-docker instead?

**Review by alangenfeld — APPROVED** [review]
i don't want to ship this pin in core so lets make sure we move if it if we can't resolve the problem

**Comment by jmsanders** [comment]
https://github.com/psf/requests/issues/6707 is our issue

**Comment by jmsanders** [comment]
I'm going to merge now that tests are passing but I want us to aim to revert/fix forward/pin more narrowly before this week's release if possible.

**Comment by jmsanders** [comment]
https://github.com/docker/docker-py/pull/3257 and our fix

**Pull request merged**